firebase-sessions/src/main/kotlin/com/google/firebase/sessions/EventGDTLogger.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.sessions
+
+import com.google.android.datatransport.*
+import com.google.android.datatransport.TransportFactory
+
+/**
+ * The [EventGDTLoggerInterface] is for testing purposes so that we can mock EventGDTLogger in other
+ * classes that depend on it.
+ *
+ * @hide
+ */
+internal interface EventGDTLoggerInterface {
+  fun log(sessionEvent: SessionEvent)
+}
+
+/**
+ * The [EventGDTLogger] is responsible for encoding and logging events to the Google Data Transport
+ * library.
+ *
+ * @hide
+ */
+internal class EventGDTLogger(private val transportFactory: TransportFactory) :
+  EventGDTLoggerInterface {
+
+  // Logs a [SessionEvent] to FireLog
+  override fun log(sessionEvent: SessionEvent) {
+    transportFactory
+      .getTransport(
+        EventGDTLogger.AQS_LOG_SOURCE,
+        SessionEvent::class.java,
+        Encoding.of("json"),
+        this::encode,
+      )
+      .send(Event.ofData(sessionEvent))
+  }
+
+  private fun encode(value: SessionEvent): ByteArray {
+    return SessionEvents.SESSION_EVENT_ENCODER.encode(value).toByteArray()
+  }
+
+  companion object {
+    // TODO What do we put for the AQS Log Source
+    private const val AQS_LOG_SOURCE = "AQS_LOG_SOURCE"
```

Comment:
What do we put for the log source? It seems to be a number on iOS, but a string on Android

Reply:
I think it needs both LogSourceName and LogSourceNumber. Firelog reference https://github.com/firebase/firebase-android-sdk/blob/master/transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/internal/LogRequest.java#L38-L42.

This is how we did it Fireperf. https://github.com/firebase/firebase-android-sdk/blob/master/firebase-perf/src/main/java/com/google/firebase/perf/config/ConfigurationConstants.java#L581-L590

Reply:
I see a lot of other SDKs just using the log source name, so I'm trying that out. Gonna send some events in the test app hopefully